Options fixes.

Satharis opened the pull request:
Added a new WindowDialog control that allows continue/save, continue/discard and cancel when closing the options menu, also fixed the other bug.

Fixes #1592 
Fixes #1593

Kasterisk reviewed (approved):
Seems good, did a quick test and it also works nicely

hhyyrylainen commented on src/general/OptionsMenu.cs:
```diff
@@ -516,17 +516,42 @@ private void OnDefaultsPressed()
+        // Save the new settings to the config file.
+        if (!Settings.Save())
+        {
+            GD.PrintErr("Failed to save new options menu settings.");
```
It would be nice to show this error in a popup box so the user isn't confused in the unlikely scenario that the saving fails.

hhyyrylainen commented on src/general/OptionsMenu.tscn:
```diff
@@ -21,10 +21,16 @@ font_data = ExtResource( 3 )
 
 [sub_resource type="StyleBoxEmpty" id=4]
 
+[sub_resource type="StyleBoxEmpty" id=9]
```
It seems that whenever I reopen this in godot editor (after resetting changes) godot absolutely wants to reorder things.
So could you close your godot editor, and reopen it, it should do some reordering on this file.

hhyyrylainen reviewed (changes requested):
I noticed that when the chromatic aberration box is checked, it doesn't react to mouse hover the same way for example the vsync option does when it is selected. When chromatic aberration isn't selected it reacts to mouse hover. Any idea what's up with that?

Satharis commented:
Added the error popup dialog and fixed that issue with mouseover, it didn't have a style set that the other checkboxes did for the case of it being pressed.

hhyyrylainen commented:
Could you move the checkbox stuff to the thrive main theme file (src/gui_common/thrive_theme.tres)? So that it doesn't need to be manually added for all checkboxes.

Satharis commented:
I checked and the style is already set in the theme actually, seems to be an engine bug.
https://github.com/godotengine/godot/issues/34727
Doesn't actually apply the style unless custom override is set on the control.

hhyyrylainen commented:
Interesting. Hopefully that fix will get merged earlier rather than later, I'll open an issue to remember to clean things up.
https://github.com/Revolutionary-Games/Thrive/issues/1599

hhyyrylainen reviewed (approved):
Everything looks to work fine now. Didn't see any further issues in the code either.
